tst/SdpApiTest.cpp

```diff
@@ -228,6 +228,101 @@ TEST_F(SdpApiTest, setTransceiverPayloadTypes_HasRtxType) {
+    PKvsPeerConnection KvsPeerConnection = (PKvsPeerConnection) (offerPc);
+
+    RtcMediaStreamTrack track;
+    PRtcRtpTransceiver pTransceiver;
+    PRtcRtpTransceiverInit pRtcRtpTransceiverInit;
+    pRtcRtpTransceiverInit = (PRtcRtpTransceiverInit) MEMCALLOC(1, SIZEOF(RtcRtpTransceiverInit));
+    pRtcRtpTransceiverInit->direction = RTC_RTP_TRANSCEIVER_DIRECTION_SENDRECV;
+
+    MEMSET(&track, 0x00, SIZEOF(RtcMediaStreamTrack));
+
+    KvsPeerConnection->isOffer = TRUE;
+
+    track.kind = MEDIA_STREAM_TRACK_KIND_VIDEO;
+    track.codec = RTC_CODEC_H264_PROFILE_42E01F_LEVEL_ASYMMETRY_ALLOWED_PACKETIZATION_MODE;
+    STRCPY(track.streamId, "myKvsVideoStream");
+    STRCPY(track.trackId, "myTrack");
+
+    EXPECT_EQ(STATUS_SUCCESS, addTransceiver(offerPc, &track, pRtcRtpTransceiverInit, &pTransceiver));
+    EXPECT_EQ(STATUS_SUCCESS, createOffer(offerPc, &sessionDescriptionInit));
+    EXPECT_PRED_FORMAT2(testing::IsSubstring, "sendrecv", sessionDescriptionInit.sdp);
+
+    freePeerConnection((PRtcPeerConnection*) KvsPeerConnection);
+}
+
+TEST_F(SdpApiTest, populateSingleMediaSection_TestTxSendOnly) {
+
+    PRtcPeerConnection offerPc = NULL;
+    RtcConfiguration configuration;
+    RtcSessionDescriptionInit sessionDescriptionInit;
+
+    // Create peer connection
+    EXPECT_EQ(createPeerConnection(&configuration, &offerPc), STATUS_SUCCESS);
+    PKvsPeerConnection KvsPeerConnection = (PKvsPeerConnection) (offerPc);
+
+    RtcMediaStreamTrack track;
+    PRtcRtpTransceiver pTransceiver;
+    PRtcRtpTransceiverInit pRtcRtpTransceiverInit;
+    pRtcRtpTransceiverInit = (PRtcRtpTransceiverInit) MEMCALLOC(1, SIZEOF(RtcRtpTransceiverInit));
+    pRtcRtpTransceiverInit->direction = RTC_RTP_TRANSCEIVER_DIRECTION_SENDONLY;
+
+    MEMSET(&track, 0x00, SIZEOF(RtcMediaStreamTrack));
+
+    KvsPeerConnection->isOffer = TRUE;
+
+    track.kind = MEDIA_STREAM_TRACK_KIND_VIDEO;
+    track.codec = RTC_CODEC_H264_PROFILE_42E01F_LEVEL_ASYMMETRY_ALLOWED_PACKETIZATION_MODE;
+    STRCPY(track.streamId, "myKvsVideoStream");
+    STRCPY(track.trackId, "myTrack");
+
+    EXPECT_EQ(STATUS_SUCCESS, addTransceiver(offerPc, &track, pRtcRtpTransceiverInit, &pTransceiver));
+    EXPECT_EQ(STATUS_SUCCESS, createOffer(offerPc, &sessionDescriptionInit));
+    EXPECT_PRED_FORMAT2(testing::IsSubstring, "sendonly", sessionDescriptionInit.sdp);
+
+    freePeerConnection((PRtcPeerConnection*) KvsPeerConnection);
+}
+
+TEST_F(SdpApiTest, populateSingleMediaSection_TestTxRecvOnly) {
+
+    PRtcPeerConnection offerPc = NULL;
+    RtcConfiguration configuration;
+    RtcSessionDescriptionInit sessionDescriptionInit;
+
+    // Create peer connection
+    EXPECT_EQ(createPeerConnection(&configuration, &offerPc), STATUS_SUCCESS);
+    PKvsPeerConnection KvsPeerConnection = (PKvsPeerConnection) (offerPc);
+
+    RtcMediaStreamTrack track;
+    PRtcRtpTransceiver pTransceiver;
+    PRtcRtpTransceiverInit pRtcRtpTransceiverInit;
```

Comment:
You don't need to put this on the heap, this will fail the leak sanitizer currently

```
RtcRtpTransceiverInit rtcRtpTransceiverInit;
rtcRtpTransceiverInit->direction = RTC_RTP_TRANSCEIVER_DIRECTION_RECVONLY;

EXPECT_EQ(STATUS_SUCCESS, addTransceiver(offerPc, &track, &rtcRtpTransceiverInit, &pTransceiver));
```

Reply:
Done